qemu/tests/check_block_size.py

```diff
@@ -43,21 +44,34 @@ def run(test, params, env):
     try:
         # Get virtio block devices in guest.
         drive_serial = str(params["drive_serial_stg"])
-        drive_path = utils_misc.get_linux_drive_path(session, drive_serial)
-        if not drive_path:
-            raise error.TestError("Could not find the specified"
-                                  "virtio block device.")
-
-        drive_kname = drive_path.split("/")[-1]
-        expect_physical = int(params.get("physical_block_size_stg", 0))
-        expect_logical = int(params.get("logical_block_size_stg", 0))
+        expect_physical = int(params.get("physical_block_size_stg", 512))
+        expect_logical = int(params.get("logical_block_size_stg", 512))
         error.context("Verify physical/Logical block size", logging.info)
-        cmd = params.get("chk_phy_blk_cmd") % drive_kname
-        logging.debug("Physical block size get via '%s'" % cmd)
-        out_physical = int(session.cmd_output(cmd))
-        cmd = params.get("chk_log_blk_cmd") % drive_kname
-        logging.debug("Logical block size get via '%s'" % cmd)
-        out_logical = int(session.cmd_output(cmd))
+        if params["os_type"] == "linux":
+            drive_path = utils_misc.get_linux_drive_path(session, drive_serial)
+            if not drive_path:
+                raise error.TestError("Could not find the specified"
+                                      "virtio block device.")
+
+            drive_kname = drive_path.split("/")[-1]
+            cmd = params.get("chk_phy_blk_cmd") % drive_kname
+            logging.debug("Physical block size get via '%s'" % cmd)
+            out_physical = int(session.cmd_output(cmd))
+            cmd = params.get("chk_log_blk_cmd") % drive_kname
+            logging.debug("Logical block size get via '%s'" % cmd)
+            out_logical = int(session.cmd_output(cmd))
+        else:
+            cmd = params.get("chk_blks_cmd_windows")
+            logging.debug("Physical/Logical block size get via '%s'" % cmd)
+            out_bs = session.cmd_output(cmd, timeout=240).strip().split("\n\n")
+            for blk_info in out_bs:
+                if blk_info.find(drive_serial) != -1:
```

Comment:
I think it could be more clear to use the 'in' clause, as it is a condition check and you don't need the index of the sub-string. eg.if drive_serial in blk_info